tests/Unit/Jobs/ProcessMailerTest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Tests\Unit\Jobs;
+
+use Tests\TestCase;
+use App\Jobs\ProcessMailer;
+use App\Mail\RemindStudent;
+use Illuminate\Support\Facades\Mail;
+use App\Models\User;
+
+class ProcessMailerTest extends TestCase
+{
+    protected $user;
+    protected $job;
+
+    public function setUp() : void
+    {
+        parent::setUp();
+        $this->user = factory(User::class)->make();;
+        $this->job = new ProcessMailer($this->user);
+    }
+
+    public function tearDown() : void
+    {
+        unset($this->user);
+        unset($this->job);
+        parent::tearDown();
+    }
+
+    public function test_handle_send_mail()
+    {
+        Mail::fake();
+        $this->job->handle();
+        Mail::assertSent(RemindStudent::class);
+        Mail::assertSent(RemindStudent::class, function($mail) {
+            return $mail->hasTo($this->user);
+        });
+    }
```

Comment:
giải thích function này

Reply:
`Mail::fake();`: Đoạn này em đùng để fake Mail (sau này tiến hành gửi và assert)
`$this->job->handle();`: em tiến hành thực hiện job (function handle() trong job là em tiến hành gửi Mail)
`Mail::assertSent(RemindStudent::class);`: em kiểm tra xem mail được gửi đi chưa
`Mail::assertSent(RemindStudent::class, function($mail) {
            return $mail->hasTo($this->user);
        });`: em kiểm tra xem thư  gửi đi có được gửi đến đúng $user hay không vì trong hàm setUp() em có tạo Job: `$this->user = factory(User::class)->make();
        $this->job = new ProcessMailer($this->user);`